numpy/_core/tests/test_multiarray.py

```diff
@@ -8712,12 +8720,19 @@ class TestArrayAttributeDeletion:
     def test_multiarray_writable_attributes_deletion(self):
         # ticket #2046, should not seqfault, raise AttributeError
         a = np.ones(2)
-        attr = ['shape', 'strides', 'data', 'dtype', 'real', 'imag', 'flat']
+        attr = ['shape', 'data', 'dtype', 'real', 'imag', 'flat']
         with suppress_warnings() as sup:
             sup.filter(DeprecationWarning, "Assigning the 'data' attribute")
             for s in attr:
                 assert_raises(AttributeError, delattr, a, s)
 
+        attr = ['strides']
+        with suppress_warnings() as sup:
+            sup.filter(DeprecationWarning, "Assigning the 'data' attribute")
+            for s in attr:
+                with pytest.warns(DeprecationWarning):
+                    assert_raises(AttributeError, delattr, a, s)
```

Comment:
this is overly complicated (probably including that filter above), because you are giving the warning when it'll raise anyway.
You should just move the warning to _after_ the error for deleting (value is `NULL`).

(Or actually, move that error to the very beginning, since we are not bound by C89 style "declarations first" anymore.)